src/android/PurchasesPlugin.java

```diff
@@ -53,6 +50,14 @@ public void onReceived(@NonNull CustomerInfo customerInfo) {
         callbackContext.sendPluginResult(result);
     }
 
+    @PluginAction(thread = ExecutionThread.UI, actionName = "configure", isAutofinish = false)
```

Comment:
Not sure if you've checked, but I just realized that maybe Cordova code executes in `ExecutionThread.MAIN` and that's where the thread change comes from

Reply:
So the `cordova-annotated-plugin-android` would not do any thread change if you set it to `ExecutionThread.MAIN` https://github.com/appfeel/cordova-annotated-plugin-android/blob/master/src/android/CordovaPluginAction.java#L71. 

Maybe that's what we need to use? Did you see `configure` being async also in iOS?